### PR TITLE
[WIP] support finality-proof sub-protocol

### LINKED ISSUE #1271: support /finality-proof/1 sub-protocol
## Expected Behavior

- see spec 4.2.3. Finality Package
-  `FinalityProofRequest` and `FinalityProofResponse` are sent over this sub-protocol

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [ ] I am planning to submit a pull request to fix this issue myself

### PR DESCRIPTION
**Pull request by anthdm**
## Changes

- Added finality-proof as sub-protocol
- Added FinalityProofRequest message
- Added FinalityProofResponse message

## Tests

```

```

## Checklist

- [ ] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [ ] I have provided as much information as possible and necessary
- [ ] I have reviewed my own pull request before requesting a review
- [ ] All integration tests and required coverage checks are passing

## Issues

- Closes #1271

### CODE REVIEW

File: dot/network/finality_proof.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	Request []byte
+}
+
+// SubProtocol returns the finality-proof sub-protocol.
+func (f *FinalityProofRequest) SubProtocol() string {
+	return finalityProofID
+}
+
+// Decode the message into a FinalityProofRequest.
+func (f *FinalityProofRequest) Decode(b []byte) error {
```
Comment (noot) [inline dot/network/finality_proof.go:38]: I think encode and decode for these types use protobuf, but I just took a look in substrate and couldn't find the proto files. I asked the spec team, will update you on this

Comment (anthdm) [reply dot/network/finality_proof.go:38]: I will let this scale decoding sit until we have more information from the spec team about this protocol.

File: dot/network/service.go [hunk 2]
```diff
@@ -379,6 +383,54 @@ func (s *Service) SendMessage(msg NotificationsMessage) {
...
+	// TODO: handle finality-proof request.
+	logger.Info("FinalityProof message %s", req)
+
+	resp := FinalityProofResponse{}
+	if err := s.host.send(peer, finalityProofID, &resp); err != nil {
+		logger.Error("failed to send FinalityProofResponse message", "peer", peer, "err", err)
+		return err
+	}
+
+	return nil
```
Comment (noot) [inline dot/network/service.go:431]: could you update this to check whether it's `FinalityProofRequest` or `FinalityProofResponse` and log it accordingly? you don't need to handle anything at the moment

**Comment by noot** [comment]
unfortunately this protocol is no longer used, so closing for now

**Pull request closed without merging**